lib/reply.js

```diff
@@ -244,7 +244,7 @@ Reply.prototype.header = function (key, value = '') {
     }
 
     if (Array.isArray(value)) {
-      this[kReplyHeaders][key].push(...value)
+      this[kReplyHeaders][key] = this[kReplyHeaders][key].concat(value)
```

Comment:
concat is usually slower. As you have two arrays in memory.
I think the issue comes from the spread operator

How is the perf if you do this?
```suggestion
      Array.prototype.push.apply(this[kReplyHeaders][key], value)
```

Reply:
value is an array here, we can't do it

Reply:
Sure we can.

const a = [1,2,3]
const b = [3,4,5]
Array.prototype.push.apply(a, b)
console.log(a) //  [1, 2, 3, 3, 4, 5]

Reply:
`apply` ftw

Reply:
```
concat x 10,043,689 ops/sec ±1.14% (84 runs sampled)
push x 47,494,392 ops/sec ±0.61% (91 runs sampled)
Fastest is push
```

```
'use strict'
const Benchmark = require('benchmark')

const suite = new Benchmark.Suite()

suite.add('concat', function () {
  ['cookie'].concat(['cookie2', 'cookie3'])
})

suite.add('push', function () {
  Array.prototype.push.apply(['cookie'], ['cookie2', 'cookie3'])
})

// add listeners
suite.on('cycle', function (event) {
  console.log(String(event.target))
})
  .on('complete', function () {
    console.log('Fastest is ' + this.filter('fastest').map('name'))
  })
  // run async
  .run()

```